src/socket/tcp.rs

```diff
@@ -757,7 +757,12 @@ impl<'a> Socket<'a> {
         }
 
         if self.is_open() {
-            return Err(ListenError::InvalidState);
+            // If we were already listening, to same endpoint, there is nothing to do.
```

Comment:
Maybe add an explanation of why this particular behavior is desirable, based on the PR description?

Reply:
Done :)